crates/revm/src/executor.rs

```diff
@@ -378,23 +386,22 @@ where
                         success: result.is_success(),
                         cumulative_gas_used,
                         logs,
-                        deposit_nonce: None,
+                        // Deposit nonce is only recorded after Regolith for deposit transactions.
+                        deposit_nonce: if self
+                            .chain_spec
+                            .is_fork_active_at_timestamp(Hardfork::Regolith, block.timestamp) &&
+                            transaction.is_deposit()
+                        {
+                            Some(old_sender_info.nonce)
+                        } else {
+                            None
+                        },
```

Comment:
I _think_ this nit should work - don't know if wrapping in parenthesis will compile nice
[`.then(|| ...)`](https://doc.rust-lang.org/std/primitive.bool.html#method.then) style nit to make this more idiomatic
```suggestion
                        deposit_nonce: (self
                            .chain_spec
                            .is_fork_active_at_timestamp(Hardfork::Regolith, block.timestamp) &&
                            transaction.is_deposit())
                            .then(|| old_sender_info.nonce),
```